### DISCUSSION TITLE
openvpn-connect seems to be broken after Terminal-App crashed during brew update/upgrade process

### DISCUSSION THREAD

**Discussion by trontur** [Homebrew/discussions#1101] [Casks] [answered]
Hi together,

my Terminal-App crashed during brew update/upgrade process.

Since then openvpn-connect seems to be broken. I can't cleanup, uninstall or reinstall. Details see below.

Thanks in advance for possible solutions!

Greetings.

brew config
HOMEBREW_VERSION: 3.0.7
ORIGIN: https://github.com/Homebrew/brew
HEAD: 59ada80ca740e72d29e625079395ee91cae4b3cf
Last commit: 6 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: b7c773411dd26509c3839a7dd17d1cc4c834247d
Core tap last commit: 26 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 4
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: quad-core 64-bit broadwell
Clang: 12.0 build 1200
Git: 2.31.0 => /usr/local/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.2.3-x86_64
CLT: 12.4.0.0.1.1610135815
Xcode: 12.4

brew doctor
=> shows no problem

brew cleanup

brew update
Already up-to-date.

brew upgrade
==> Casks with 'auto_updates' or 'version :latest' will not be upgraded; pass `--greedy` to upgrade them.
==> Upgrading 1 outdated package:
openvpn-connect 3.2.6,3136 -> 3.2.7,3220
==> Upgrading openvpn-connect
==> Downloading https://swupdate.openvpn.net/downloads/connect/openvpn-connect-3.2.7.3220_signed.dmg
Already downloaded: /Users/wrheinschmitt/Library/Caches/Homebrew/downloads/2b07f9715e3466d1217fe13870c1451f93e829e339701d2f10daaebf48f76bde--openvpn-connect-3.2.7.3220_signed.dmg
==> Removing launchctl service org.openvpn.client
Password:
==> Removing launchctl service org.openvpn.helper
==> Running uninstall script /Applications/OpenVPN Connect/Uninstall OpenVPN Connect.app/Contents/Resources/remove.sh
==> Purging files for version 3.2.7,3220 of Cask openvpn-connect
Error: openvpn-connect: uninstall script /Applications/OpenVPN Connect/Uninstall OpenVPN Connect.app/Contents/Resources/remove.sh does not exist.

brew uninstall openvpn-connect
==> Uninstalling Cask openvpn-connect
==> Removing launchctl service org.openvpn.client
==> Removing launchctl service org.openvpn.helper
==> Running uninstall script /Applications/OpenVPN Connect/Uninstall OpenVPN Connect.app/Contents/Resources/remove.sh
Error: uninstall script /Applications/OpenVPN Connect/Uninstall OpenVPN Connect.app/Contents/Resources/remove.sh does not exist.

brew reinstall openvpn-connect
==> Downloading https://swupdate.openvpn.net/downloads/connect/openvpn-connect-3.2.7.3220_signed.dmg
Already downloaded: /Users/wrheinschmitt/Library/Caches/Homebrew/downloads/2b07f9715e3466d1217fe13870c1451f93e829e339701d2f10daaebf48f76bde--openvpn-connect-3.2.7.3220_signed.dmg
==> Uninstalling Cask openvpn-connect
==> Removing launchctl service org.openvpn.client
==> Removing launchctl service org.openvpn.helper
==> Running uninstall script /Applications/OpenVPN Connect/Uninstall OpenVPN Connect.app/Contents/Resources/remove.sh
Warning: uninstall script /Applications/OpenVPN Connect/Uninstall OpenVPN Connect.app/Contents/Resources/remove.sh does not exist; skipping.
==> Uninstalling packages:
org.openvpn.client.pkg
find: /Applications/OpenVPN Connect/OpenVPN Connect.app: No such file or directory
Error: Failure while executing; `/usr/bin/sudo -E -- /usr/bin/xargs -0 -- /bin/bash -c set\ -euo\ pipefail'
''
'for\ path\ in\ \"\$\{@\}\"\;\ do'
'\ \ if\ \[\[\ -e\ \"\$\{path\}/.DS_Store\"\ \]\]\;\ then'
'\ \ \ \ /bin/rm\ -f\ \"\$\{path\}/.DS_Store\"'
'\ \ fi'
''
'\ \ \#\ Some\ packages\ leave\ broken\ symlinks\ around\;\ we\ clean\ them\ out\ before'
'\ \ \#\ attempting\ to\ \`rmdir\`\ to\ prevent\ extra\ cruft\ from\ accumulating.'
'\ \ /usr/bin/find\ \"\$\{path\}\"\ -mindepth\ 1\ -maxdepth\ 1\ -type\ l\ \!\ -exec\ /bin/test\ -e\ \{\}\ \\\;\ -delete'
''
'\ \ if\ \[\[\ -L\ \"\$\{path\}\"\ \]\]\;\ then'
'\ \ \ \ \#\ Delete\ directory\ symlink.'
'\ \ \ \ /bin/rm\ \"\$\{path\}\"'
'\ \ elif\ \[\[\ -d\ \"\$\{path\}\"\ \]\]\;\ then'
'\ \ \ \ \#\ Delete\ directory\ if\ empty.'
'\ \ \ \ /usr/bin/find\ \"\$\{path\}\"\ -maxdepth\ 0\ -type\ d\ -empty\ -exec\ /bin/rmdir\ \{\}\ \\\;'
'\ \ else'
'\ \ \ \ \#\ Try\ \`rmdir\`\ anyways\ to\ show\ a\ proper\ error.'
'\ \ \ \ /bin/rmdir\ \"\$\{path\}\"'
'\ \ fi'
'done'
' --` exited with 1. Here's the output:
find: /Applications/OpenVPN Connect/OpenVPN Connect.app: No such file or directory

**Comment by trontur** [comment, marked as answer]
Fixed it:

- installed openvpn-connect **without** using brew


- then check: `brew info openvpn-connect`
```
openvpn-connect: 3.2.7,3220
https://openvpn.net/client-connect-vpn-for-mac-os/
/usr/local/Caskroom/openvpn-connect/3.2.6,3136 (55.6MB)
From: https://github.com/Homebrew/homebrew-cask/blob/HEAD/Casks/openvpn-connect.rb
```

- uninstalled it **with using** brew: `brew uninstall openvpn-connect`

- installed it **with using** brew: `brew install openvpn-connect`
`openvpn-connect was successfully installed!`

But i wonder how it got broken in the first place...

**Reply by chrismjohnston** [reply]
Thanks, I've been having this same issue and this worked.
